File: app/src/main/java/com/example/dagger_hilt/sys/util/UtilsText.java
package com.example.dagger_hilt.sys.util;

import com.google.android.material.textfield.TextInputLayout;

public class UtilsText {
    private final TextInputLayout editText;

    public UtilsText(TextInputLayout editText) {
        this.editText = editText;
    }

    public Boolean isNotBlank() {
        var result = !editText.getEditText().getText().toString().isEmpty();
        editText.setError(result ? null : "Campo requerido");
        return result;
    }
}

File: app/src/main/java/com/example/dagger_hilt/sys/util/UtilsText.kt
package com.example.dagger_hilt.sys.util

import com.google.android.material.textfield.TextInputLayout

object UtilsText {
    fun TextInputLayout.isNotBlank(): Boolean {
        var result = false
        error = if (editText!!.text.toString().isNotBlank()) {
            result = true
            null
        } else "Campo requerido"
        return result
    }
}